segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/rolling/RollingStorageTestBase.java

```diff
@@ -123,6 +124,37 @@ public void testWriteAfterHeaderMerge() throws Exception {
         Assert.assertEquals("Unexpected number of chunks created.", 1, readHandle.chunks().size());
     }
 
+    @Test
+    public void testWriteOnRollOverBoundary() throws Exception {
+        final String segmentName = "Segment";
+        final int maxLength = 3; // Really small rolling length.
+
+        val seq1 = "01234";
+        val seq2 = "56789";
+        val totalWriteLength = seq1.length() + seq2.length();
+
+        @Cleanup
+        val s = createStorage();
+        s.initialize(1);
+
+        val writeHandle = s.create(segmentName, new SegmentRollingPolicy(maxLength), TIMEOUT)
+                .thenCompose(v -> s.openWrite(segmentName)).join();
+
+        val byteInputStream1 = new ByteArrayInputStream(seq1.getBytes());
+        val byteInputStream2 = new ByteArrayInputStream(seq2.getBytes());
+
+        val sequenceInputStream = new SequenceInputStream(byteInputStream1, byteInputStream2);
+
+        // This write should cause 3 rollovers.
+        s.write(writeHandle, 0, sequenceInputStream, totalWriteLength, TIMEOUT).join();
```

Comment:
Can the 3 rollovers be reflected in the test code?

Reply:
We write 10 bytes, with max rolling length set to 3.

Reply:
Added additional checks.